go-controller/pkg/cni/helper_linux.go

```diff
@@ -728,18 +728,39 @@ func (*defaultPodRequestInterfaceOps) UnconfigureInterface(pr *PodRequest, ifInf
 	if !ifInfo.IsDPUHostMode {
 		var err error
 		// host side interface deletion
-		hostIfName := pr.SandboxID[:(15-len(ifnameSuffix))] + ifnameSuffix
+		var hostIfName string
+		if !util.IsNetworkSegmentationSupportEnabled() || isSecondary {
+			// this is a secondary network (not primary) or segmentation is not enabled
+			hostIfName = pr.SandboxID[:(15-len(ifnameSuffix))] + ifnameSuffix
+		}
 		if pr.CNIConf.DeviceID != "" {
 			hostIfName, err = util.GetFunctionRepresentorName(pr.CNIConf.DeviceID)
 			if err != nil {
 				klog.Errorf("Failed to get the representor name for DeviceID %s for pod %s: %v",
 					pr.CNIConf.DeviceID, podDesc, err)
 			}
 		}
+		portList, err := ovsFind("interface", "name", "external-ids:sandbox="+pr.SandboxID)
+		if err != nil {
+			return fmt.Errorf("failed to list interfaces in OVS during delete for sandbox: %s, err: %w",
+				pr.SandboxID, err)
+		}
+		// hostIfName is not empty if using device ID, a secondary network, or segmentation not enabled
+		// delete the port in traditional fashion
 		if hostIfName != "" {
-			pr.deletePorts(hostIfName, pr.PodNamespace, pr.PodName)
+			pr.deletePort(hostIfName, pr.PodNamespace, pr.PodName)
+		} else {
```

Comment:
no, we will hit this if network segmentation is enabled and we are deleting the primary CNI. At that point we have to search for multiple interfaces in case this primary CNI is serving a primary UDN.

Reply:
hmm anytime net-seg is enabled this will be hit right? doesn't matter if we are deleting primary or secondary here right?